framework/src/systems/SystemBase.C

```diff
@@ -1356,7 +1356,9 @@ SystemBase::solutionState(const unsigned int state,
                " was requested in ",
                name(),
                " but only up to state ",
-               _solution_states[static_cast<unsigned short>(iteration_type)].size() - 1,
+               (_solution_states[static_cast<unsigned short>(iteration_type)].size() == 0)
```

Comment:
Do we not always have at least one solution state requested?

Reply:
Turns out not. The logs were showing me -1

Reply:
But as an unsigned ont

Reply:
I’d be curious to get a stack trace for when it’s -1

Reply:
Yeah, this confuses me too. Is NS just doing something super early?

Shouldn't we just request state 0 in the system on construction? and add an assertion here that size > 0

Reply:
> I’d be curious to get a stack trace for when it’s -1

the print is size() - 1. So it's 0, but -1 in the print

Reply:
Basically we left it up to the user to request the previous NL state. 

I tried to do it programmatically but it's not working. So instead it's being done on the fly in a non thread safe way. Hence the crash

This PR is trying to fix that. I did a better take on it in the linear FV physics PR.

Otherwise the simplest fix of all is to add the parameter in the input file to request the old nonlinear state

Reply:
Can you give a CIVET URL for the failure? There's no identification of this issue on the issue number you reference. @loganharbour and I are asking when/why this happens. Because we `initSolutionState` during `initialSetup` then NS must be requesting solution states directly early on before `initalSetup`

Reply:
[it exodiffed here](https://civet.inl.gov/job/2468035/)

Reply:
you can also reproduce this locally most likely, with -p2 --n-threads=2 --distributed-mesh

Reply:
created an issue with the civet URL and a description of what happened